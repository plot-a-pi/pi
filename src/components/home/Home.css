.Home {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}