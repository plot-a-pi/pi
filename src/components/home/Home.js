import React from 'react';
import ButtonNav from './ButtonNav';
import CircumferenceVsDiameterWrapper from '../graphs/CicumferenceVsDiameterWrapper';
import GlobalPiVsCountGraph from '../graphs/GlobalPiVsCountGraph';
import styles from './Home.css';
import { style } from 'd3';

const Home = () => {

  return (
    <div className={styles.home}>
      <section className={styles.graphs}>
        <CircumferenceVsDiameterWrapper />
        <div className={styles.global}>
          <GlobalPiVsCountGraph />
        </div>
      </section>
      <div className={styles.buttons}>
        <ButtonNav />
      </div>
    </div>
  );
};

export default Home;
