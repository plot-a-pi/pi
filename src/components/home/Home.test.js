import React from 'react';
import { shallow } from 'enzyme';
import Home from './Home';

jest.mock('react-socket-io-hooks', () => ({
  useEmitEvent: () => {},
  useSocketState: () => ({
    stats:  {
      piApproximationArray: [4, 3, 3.5]
    }
  }),
  useSocket: () => ({
    socket: {
      connected: null
    }
  }),
}));

describe('Stats component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
});
