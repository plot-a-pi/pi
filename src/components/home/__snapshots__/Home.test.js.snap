// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component should match snapshot 1`] = `
<div
  className="Home"
>
  <CvsDScatterplot
    className="graphA"
    line={true}
    stats={
      Object {
        "piApproximationArray": Array [
          4,
          3,
          3.5,
        ],
      }
    }
    title="Global Circle Measurements"
  />
  <CvDGraphStats
    className={
      Object {
        "piApproximationArray": Array [
          4,
          3,
          3.5,
        ],
      }
    }
    stats={
      Object {
        "piApproximationArray": Array [
          4,
          3,
          3.5,
        ],
      }
    }
  />
  <PivsCountScatterplot
    className="graphB"
    data={
      Array [
        Array [
          1,
          4,
        ],
        Array [
          2,
          3,
        ],
        Array [
          3,
          3.5,
        ],
      ]
    }
    title="Global Pi Approximation vs Count"
    xLabel="Count"
    yLabel="Pi Approximation"
  />
  <PiCrawler
    className="piCrawler"
  />
</div>
`;
